thriftbp/ttl_client.go

```diff
@@ -17,50 +22,135 @@ const DefaultMaxConnectionAgeJitter = 0.1
 
 var _ Client = (*ttlClient)(nil)
 
+type ttlClientLockedData struct {
```

Comment:
🔕 (nit, naming) I typically call it "fooState" -- so in this case "ttlClientState"

and then the field I typically name "state"

but the names you have are perfectly readable, so 🤷